### PR TITLE
Reports MLMethods updates

### PR DESCRIPTION
**Pull request by BlueBasilisk**
Contains updates from 'parallel_reports' branch (which you recently reviewed) + another branch:
- Minor YAML change: move number_of_processes in ExploratoryAnalysis instruction definition one level up; outside of 'analyses'
- make number_of_processes available to Reports, allowing for parallelization
- HTML readability: Added 'info' to ReportResult, describing the report in one or few sentences. Display this in the generated HTML. Also display generated 'output text' in HTML (prior we only displayed figures and tables), always show the report name (key) in the HTML, and minor other refactoring for general HTML readability
- small update: changed feature delimiter from /// to - for readability of feature names (only used by Kmer encoder)
- small fix: throw no error if empty list is given to signify 'no reports' in TrainMLModelInstruction yaml
- pass Label object instead of label string to MLMethods and Reports, also, to avoid confusion: consistently use the variable name 'label' to mean Label object, and 'label_str' to mean the string, and add the type (str/Label) in the signature of many methods
- Utility functionality in LabelConfiguration: automatically set the 'postive_class' if the labels of a binary class are 0/1, true/false or positive/negative
- cleaned up __init__ of all reports, pass all relevant parameters to parent reports. 
- remove some unused methods of MLMethod and subclasses

not entirely sure if we should increment the patch or minor version number?

### CODE REVIEW

File: immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def build(report_results: dict, result_path: Path, instruction_result_paths: dic
 
     @staticmethod
     def _make_html_map(report_results: dict, result_path: Path, instruction_result_paths: dict) -> dict:
+        print(report_results)
```
Comment (pavlovicmilena) [inline immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py:26]: could you remove this? :)

File: immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py [hunk 2]
```diff
@@ -35,4 +36,6 @@ def _make_html_map(report_results: dict, result_path: Path, instruction_result_p
         if len(html_map['reports']) == 0:
             html_map['show_reports'] = False
 
+        print(html_map)
```
Comment (pavlovicmilena) [inline immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py:39]: and other print statements :)

File: immuneML/workflows/steps/MLMethodAssessment.py [hunk 3]
```diff
@@ -27,22 +27,22 @@ def run(input_params: MLMethodAssessmentParams = None):
...
+                                           label_name=input_params.label.name, split_index=input_params.split_index, predicted_y=predicted_y,
                                            predicted_proba_y=predicted_proba_y, true_y=true_y, method=input_params.method,
                                            ml_score_path=input_params.ml_score_path)
 
         return scores
 
-    @staticmethod
-    def _get_optimization_metric(df: pd.DataFrame, label: str, metric: Metric) -> float:
-        return df["{}_{}".format(label, metric.name.lower())].iloc[0]
+    # @staticmethod
```
Comment (pavlovicmilena) [inline immuneML/workflows/steps/MLMethodAssessment.py:40]: why is this commented out? if not used, remove it? :)

File: test/ml_methods/test_sklearnMethod.py [hunk 4]
```diff
@@ -59,3 +60,19 @@ def test_store(self):
         self.assertTrue(isinstance(svm2, SVC))
 
         shutil.rmtree(path)
+    #
```
Comment (pavlovicmilena) [inline test/ml_methods/test_sklearnMethod.py:63]: ? :)

File: immuneML/reports/train_ml_model_reports/ReferenceSequenceOverlap.py [hunk 5]
```diff
@@ -71,37 +72,36 @@ def build_object(cls, **kwargs):
...
+                df_filename = self.result_path / f"assessment_split_{assessment_state.split_index + 1}_overlap_sequences_{self.label.name}"
                 figure, data = self._compute_model_overlap(figure_filename, df_filename, encoder,
                                                            f"overlap sequences between the model for assessment split "
                                                            f"{assessment_state.split_index + 1} and reference list")
                 figures.append(figure)
                 tables.append(data)
 
-        return ReportResult(self.name, output_figures=figures, output_tables=tables)
+        return ReportResult(self.name,
+                            info="A Venn diagram between the list of disease-associated sequences produced by the SequenceAbundance encoder and a list of reference receptor sequences, and a file containing the overlapping sequences.",
```
Comment (pavlovicmilena) [inline immuneML/reports/train_ml_model_reports/ReferenceSequenceOverlap.py:103]: could you make sure that this info is in the same place for all reports? now it is sometimes an object variable, sometimes it's a constant hard-coded when constructing ReportResult object :)

File: immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py (hunk 1 above)
Comment (BlueBasilisk) [reply immuneML/presentation/html/MultiDatasetBenchmarkHTMLBuilder.py:26]: ooops 😱  yes will remove all the prints

File: test/ml_methods/test_sklearnMethod.py (hunk 4 above)
Comment (BlueBasilisk) [reply test/ml_methods/test_sklearnMethod.py:63]: fixed the bug that made this commented out test fail + added the test

**Review by pavlovicmilena — APPROVED** [review]

**Pull request merged**